service/history/timerBuilder.go

```diff
@@ -226,47 +201,45 @@ func (tb *timerBuilder) LoadUserTimers(msBuilder *mutableStateBuilder) {
 }
 
 // IsTimerExpired - Whether a timer is expired w.r.t reference time.
-func (tb *timerBuilder) IsTimerExpired(td *timerDetails, referenceTime int64) bool {
-	expiry, _ := DeconstructTimerKey(td.SequenceID)
-	return expiry <= referenceTime
+func (tb *timerBuilder) IsTimerExpired(td *timerDetails, referenceTime time.Time) bool {
+	// Cql timestamp is in milli sec resolution, here we do the check in terms of second resolution.
+	expiry := td.SequenceID.VisibilityTimestamp.Unix()
+	return expiry <= referenceTime.Unix()
 }
 
 // createDecisionTimeoutTask - Creates a decision timeout task.
 func (tb *timerBuilder) createDecisionTimeoutTask(fireTimeOut int32, eventID int64) *persistence.DecisionTimeoutTask {
-	expiryTime := common.AddSecondsToBaseTime(time.Now().UnixNano(), int64(fireTimeOut))
-	seqID := ConstructTimerKey(expiryTime, tb.seqNumGen.NextSeq())
+	expiryTime := time.Now().Add(time.Duration(fireTimeOut) * time.Second)
```

Comment:
Instead of using time.Now here can we use an interface so it is easy to mock system clock.